pkg/controller/atlasdatabaseuser/atlasdatabaseuser_controller.go

```diff
@@ -123,18 +124,16 @@ func (r *AtlasDatabaseUserReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return result.ReconcileResult(), nil
 	}
 
-	atlasClient, orgID, err := r.AtlasProvider.Client(ctx, project.ConnectionSecretObjectKey(), log)
+	dus, err := dbuser.NewAtlasDatabaseUsersService(ctx, r.AtlasProvider, project.ConnectionSecretObjectKey(), log)
 	if err != nil {
 		result = workflow.Terminate(workflow.AtlasAPIAccessNotConfigured, err.Error())
 		workflowCtx.SetConditionFromResult(api.DatabaseUserReadyType, result)
 
 		return result.ReconcileResult(), nil
 	}
-	workflowCtx.OrgID = orgID
```

Comment:
we use `workflow.Context` everywhere and while DBUser doesn't need to be aware of orgID, I'm concerned that leaving the context incomplete might lead to bugs in the feature. Maybe worth adding a comment saying that it won't be always available once it will probably be moved out of context

Reply:
I can add a comment indeed. I removed as I do not think it is used or required, will double check it thought.

Also `workflow.Context` is a bag of things and not all controllers need everything in such a bag... IMO we should get rid of it when we can.

Reply:
Note added, BTW going forward is org id is used, it should be part of the internal type addendums, like the secret or project id.